crates/evm/db.rs

```diff
@@ -1,25 +1,28 @@
-use ethereum_rust_core::{Address as CoreAddress, H256 as CoreH256};
+use ethereum_rust_core::{types::BlockNumber, Address as CoreAddress, H256 as CoreH256};
 use ethereum_rust_storage::{error::StoreError, Store};
 use revm::primitives::{
     AccountInfo as RevmAccountInfo, Address as RevmAddress, Bytecode as RevmBytecode,
     Bytes as RevmBytes, B256 as RevmB256, U256 as RevmU256,
 };
 
-pub struct StoreWrapper(pub Store);
+pub struct StoreWrapper {
+    pub store: Store,
+    pub block_number: BlockNumber,
```

Comment:
I think it makes more sense to store a `block_hash` here since we might want to execute a block that is not from the canonical chain. Thoughts?

Reply:
We only use this field to fetch `account_info` from the store, which is indexed by `block_number`, so changing the field to block hash wouldn't fix the issue.
I would leave it to a separate refactor